src/h2_streams.erl

```diff
@@ -0,0 +1,541 @@
+-module(h2_streams).
+-include("http2.hrl").
+-include("h2_streams.hrl").
+
+-export(
+   [
+    new/1,
+    get/2,
+    close/4,
+    initiate_stream/7,
+    peer_initiated_stream/7,
+    replace/2,
+    send_what_we_can/4,
+    sort/1,
+    notify_pid/1,
+    update_all_recv_windows/2,
+    update_all_send_windows/2,
+    update_peer_max_active/2,
+    update_self_max_active/2
+   ]).
+
+-spec new(
+        client | server
+       ) -> streams().
+new(client) ->
```

Comment:
Would documentation be enough here? `initiate_stream/7` and `peer_initiated_stream/7` are what actually create streams and add them to the set, and I feel like those are well named. Something like `init` makes me think of gen_server callbacks, and not just a simple data structure. This was meant to mimic something like `dict:new()`